---
title: "Перехват исключения (стоит ли) C#"
se.owner.user_id: 
se.owner.display_name: "user358283"
se.owner.link: ""
se.link: "https://ru.stackoverflow.com/questions/1046558/%d0%9f%d0%b5%d1%80%d0%b5%d1%85%d0%b2%d0%b0%d1%82-%d0%b8%d1%81%d0%ba%d0%bb%d1%8e%d1%87%d0%b5%d0%bd%d0%b8%d1%8f-%d1%81%d1%82%d0%be%d0%b8%d1%82-%d0%bb%d0%b8-c"
se.question_id: 1046558
se.post_type: question
se.score: 0
---
<p>Мне кажется код немного кривоват, но не могу понять как лучше его переделать.
Код:</p>

<pre><code>void Button1Click(object sender, EventArgs e)
{
    if (openFileDialog1.ShowDialog() == DialogResult.OK) 
    {
        if (!System.IO.File.Exists(openFileDialog1.FileName))
        {
            throw new PathException("Некорректный путь к файлу", openFileDialog1.FileName);
        }
        try
        {
            textBox7.Text = openFileDialog1.FileName;
        }
        catch (PathException ex)
        {
            MessageBox.Show(String.Format("{0}: {1}", ex.Message, ex.PathValue), "Ошибка.", MessageBoxButtons.OK, MessageBoxIcon.Error, MessageBoxDefaultButton.Button1);
        }
    }       
}
</code></pre>

<p>Класс исключения:</p>

<pre><code>public class PathException : ArgumentException
{   
    public string PathValue {get; set;}
    public PathException(string message, string pathValue)
        :base(message)
    {
        PathValue = pathValue;
    }
}
</code></pre>

<p>Что можно исправить? </p>
